app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt

```diff
@@ -120,43 +126,49 @@ class NewVersionWorker(
     }
 
     override fun doWork(): Result {
-        try {
+        return try {
             checkNewVersion()
+            Result.success()
         } catch (e: IOException) {
             Log.w(TAG, "Could not fetch NewPipe API: probably network problem", e)
-            return Result.failure()
+            Result.failure()
         } catch (e: ReCaptchaException) {
             Log.e(TAG, "ReCaptchaException should never happen here.", e)
-            return Result.failure()
+            Result.failure()
         }
-        return Result.success()
     }
 
     companion object {
         private val DEBUG = MainActivity.DEBUG
         private val TAG = NewVersionWorker::class.java.simpleName
         private const val NEWPIPE_API_URL = "https://newpipe.net/api/data.json"
+        private const val IS_MANUAL = "isManual"
 
         /**
-         * Start a new worker which
-         * checks if all conditions for performing a version check are met,
-         * fetches the API endpoint [.NEWPIPE_API_URL] containing info
-         * about the latest NewPipe version
-         * and displays a notification about ana available update.
+         * Start a new worker which checks if all conditions for performing a version check are met,
+         * fetches the API endpoint [.NEWPIPE_API_URL] containing info about the latest NewPipe
+         * version and displays a notification about an available update if one is available.
          * <br></br>
-         * Following conditions need to be met, before data is request from the server:
+         * Following conditions need to be met, before data is requested from the server:
          *
          *  *  The app is signed with the correct signing key (by TeamNewPipe / schabi).
          * If the signing key differs from the one used upstream, the update cannot be installed.
          *  * The user enabled searching for and notifying about updates in the settings.
          *  * The app did not recently check for updates.
          * We do not want to make unnecessary connections and DOS our servers.
-         *
          */
         @JvmStatic
-        fun enqueueNewVersionCheckingWork(context: Context) {
-            val workRequest: WorkRequest =
-                OneTimeWorkRequest.Builder(NewVersionWorker::class.java).build()
+        fun enqueueNewVersionCheckingWork(context: Context, isManual: Boolean) {
+            // Search for updates immediately when update checks are enabled. Reset the expire time
+            // if the check was manual. This is necessary to check for an update immediately.
+            if (isManual) {
+                PreferenceManager.getDefaultSharedPreferences(context).edit {
+                    putLong(context.getString(R.string.update_expiry_key), 0)
+                }
+            }
```

Comment:
This is not what I meant: remove this and just add an `if (!manual)` around [these lines](https://github.com/TeamNewPipe/NewPipe/blob/6ca5b4108ba98274ab7602bbbb1ebee03439147a/app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt#L81-L87)

Reply:
Is the change okay?